### ISSUE TITLE
[MNG-6160] Forking process not always aware of reactor projects

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7284]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** opened **[MNG-6160](https://issues.apache.org/jira/browse/MNG-6160?redirect=false)** and commented

While running `mvn clean -e -Preporting site` on the Maven Artifact Resolver 1.0.3-SNAPSHOT I noticed downloads of files which should be part of the reactor.

```
[INFO] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[INFO] Forking Maven Artifact Resolver SPI 1.0.3-SNAPSHOT
[INFO] >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Downloading: https://repository.apache.org/snapshots/org/apache/maven/resolver/maven-resolver-api/1.0.3-SNAPSHOT/maven-resolver-api-1.0.3-20170122.151134-3.pom
Downloaded: https://repository.apache.org/snapshots/org/apache/maven/resolver/maven-resolver-api/1.0.3-SNAPSHOT/maven-resolver-api-1.0.3-20170122.151134-3.pom (2 KB at 1.6 KB/sec)
```

There are 2 issues with this:
- Maven is using other files than the expectd files from the reactor
- In case those files don't exist it'll fail.

See log file for complete evidence

---

**Affects:** 3.3.9

**Attachments:**
- [mvn.log](https://issues.apache.org/jira/secure/attachment/12848787/mvn.log) (_150.60 kB_)

**Issue Links:**
- [MNG-4660](https://issues.apache.org/jira/browse/MNG-4660) Use of --resume-from in multi-module project fails with missing inter-module dependencies
 (_**"depends upon"**_)

**Backported to:** [4.0.0-alpha-2](https://github.com/apache/maven/milestone/113?closed=1)

**Comment by jira-importer** [comment]
**[Maarten Mulders](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mthmulders)** commented

I've tried to reproduce this behavior with Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f) and Maven 3.7.0-SNAPSHOT (ef8c95eb397651e10f677763dfcd9c8cea7c27b0). The Maven Artifact Resolver project has moved on from 1.0.3-SNAPSHOT to 1.4.3-SNAPSHOT.

* With 3.6.3, I can see that the SNAPSHOT dependency from SPI to API, although available in the Reactor, is resolved by downloading it from the Apache Snapshots repository. This confirms the original report.
* After that, I have removed all SNAPSHOT artifacts under `org.apache.maven.resolver` in my local repository.
* With 3.7.0-SNAPSHOT, I can see that the SNAPSHOT dependency from SPI to API is not resolved by downloading it from the Apache Snapshots repository.

I think that this issue has been resolved when we fixed MNG-4660.
